sktime/transformers/matrix_profile.py

```diff
@@ -0,0 +1,221 @@
+                min_index = k
+        mp[i] = min_value
+        ip[i] = min_index
+
+        return mp, ip
+
+
+    def stomp_self(ts, m):
+        """
+            STOMP implementation for self-similarity join.
+
+            Parameters
+            ----------
+                ts: numpy.array
+                    Time series.
+                m: int
+                    Length of the subsequences.
+
+            Output
+            ------
+                mp: numpy.array
+                    Array with the distance between every subsequence from ts1 to the nearest subsequence with same length from ts2.
+                ip: numpy.array
+                    Array with the index of the nearest neighbor of ts1 in ts2.
+        """
+
+        ts_len = ts.size
+
+        ts = ts.flatten()
+
+        # Number of subsequences
+        n_subs = ts_len-m+1
+
+        # Compute the mean and standard deviation
+        ts_mean = [np.mean(ts[i:i+m]) for i in range(0, n_subs)]
+        ts_std = [np.std(ts[i:i+m]) for i in range(0, n_subs)]
+
+        # Compute the dot products between the first subsequence and every other subsequence
+        dot_prod = sliding_dot_products(ts[0:m], ts, m, ts_len)
+        first_dot_prod = np.copy(dot_prod)
+
+        # Initialization
+        mp = np.full(n_subs, float('inf')) # matrix profile
+        ip = np.zeros(n_subs) # index profile
+
+        # Compute the distance profile for the first subsequence
+        dp = calculate_distance_profile(dot_prod, ts_mean[0], ts_std[0], ts_mean, ts_std, m, n_subs)
+
+        # Updates the matrix profile
+        mp, ip = minimum_distance(mp, ip, dp, 0, m, n_subs)
+
+        for i in range(1, n_subs):
+            for j in range(n_subs-1, 0, -1):
+                dot_prod[j] = dot_prod[j-1] - ts[j-1]*ts[i-1] + ts[j-1+m]*ts[i-1+m]  # compute the next dot products using the previous ones
+            dot_prod[0] = first_dot_prod[i]
+            dp = calculate_distance_profile(dot_prod, ts_mean[i], ts_std[i], ts_mean, ts_std, m, n_subs)
+            mp, ip = minimum_distance(mp, ip, dp, i, m, n_subs)
+
+        return mp, ip
+
+
+    def transform(self, X, m):
+        """
+            Takes as input a time series dataset and returns the matrix profile and
+            index profile for each time series of the dataset.
+
+            Parameters
+            ----------
+                X: pandas.DataFrame
+                    Time series dataset.
+                m: int
+                    Length of the subsequences.
+
+            Output
+            ------
+                mp: numpy.ndarray
+                    N-dimensional array with the matrix profile for each time series.
+                ip: numpy.ndarray
+                    N-dimensional array with the index profile for each time series.
+        """
+
+        x_size = X.size
+
+        # Convert into tabular format
+        tabulariser = Tabulariser()
+        X = tabulariser.transform(X.iloc[:, :1])
```

Comment:
After adding input checks (`check_X_is_univariate`), we can simply write:
```suggestion
        X = tabulariser.transform(X)
```